aspnetcore/security/authorization/policies.md

```diff
@@ -10,6 +10,240 @@ uid: security/authorization/policies
+
+The preceding code registers `MinimumAgeHandler` as a singleton by invoking `services.AddSingleton<IAuthorizationHandler, MinimumAgeHandler>();`. Handlers can be registered using any of the built-in [service lifetimes](xref:fundamentals/dependency-injection#service-lifetimes).
+
+## What should a handler return?
+
+Note that the `Handle` method in the [handler example](#security-authorization-handler-example) returns no value. How is a status of either success or failure indicated?
+
+* A handler indicates success by calling `context.Succeed(IAuthorizationRequirement requirement)`, passing the requirement that has been successfully validated.
+
+* A handler doesn't need to handle failures generally, as other handlers for the same requirement may succeed.
+
+* To guarantee failure, even if other requirement handlers succeed, call `context.Fail`.
+
+If a handler calls `context.Succeed` or `context.Fail`, all other handlers are still called. This allows requirements to produce side effects, such as logging, which takes place even if another handler has successfully validated or failed a requirement. When set to `false`, the [InvokeHandlersAfterFailure](/dotnet/api/microsoft.aspnetcore.authorization.authorizationoptions.invokehandlersafterfailure#Microsoft_AspNetCore_Authorization_AuthorizationOptions_InvokeHandlersAfterFailure) property short-circuits the execution of handlers when `context.Fail` is called. `InvokeHandlersAfterFailure` defaults to `true`, in which case all handlers are called.
+
+> [!NOTE]
+> Authorization handlers are called even if authentication fails.
+
+<a name="security-authorization-policies-based-multiple-handlers"></a>
+
+## Why would I want multiple handlers for a requirement?
+
+In cases where you want evaluation to be on an **OR** basis, implement multiple handlers for a single requirement. For example, Microsoft has doors which only open with key cards. If you leave your key card at home, the receptionist prints a temporary sticker and opens the door for you. In this scenario, you'd have a single requirement, *BuildingEntry*, but multiple handlers, each one examining a single requirement.
+
+*BuildingEntryRequirement.cs*
+
+[!code-csharp[](policies/samples/PoliciesAuthApp1/Services/Requirements/BuildingEntryRequirement.cs?name=snippet_BuildingEntryRequirementClass)]
+
+*BadgeEntryHandler.cs*
+
+[!code-csharp[](policies/samples/PoliciesAuthApp1/Services/Handlers/BadgeEntryHandler.cs?name=snippet_BadgeEntryHandlerClass)]
+
+*TemporaryStickerHandler.cs*
+
+[!code-csharp[](policies/samples/PoliciesAuthApp1/Services/Handlers/TemporaryStickerHandler.cs?name=snippet_TemporaryStickerHandlerClass)]
+
+Ensure that both handlers are [registered](xref:security/authorization/policies#security-authorization-policies-based-handler-registration). If either handler succeeds when a policy evaluates the `BuildingEntryRequirement`, the policy evaluation succeeds.
```

Comment:
This appears to reference itself using a full xref - is that intentional, or should I simplify it to just the `#...` part while I'm in here?

Reply:
> This appears to reference itself using a full xref - is that intentional, or should I simplify it to just the `#...` part while I'm in here?

My philosophy, if it ain't broke. I'm fine with purists who like the cleaner, simpler approach.